### PR TITLE
feat: APP-9 Add a banner at top of project pages and individual profile page on first view

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/jira/software/c/projects/ENG/boards/51?selectedIssue=APP-9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

- BannerCard component on storybook: https://deploy-preview-2336--regen-storybook.netlify.app/?path=/story/molecules-bannercard--default

- From https://deploy-preview-2336--regen-marketplace.netlify.app/:
1. Login
2. Visit your own profile page, you should see the banner at the top. If you navigate away and navigate back to this page, it won't be there anymore
3. Visit a few of your project pages, you should see the project banner at the top when first viewing a given project page

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by erikalogie** [comment]
> I think there was a small miscommunication with the requirements for this task. I just tested the functionality and it behaves exactly as described in the description above, but I think it would actually be preferable to have a slightly different functionality.
> 
> Suggested functionality changes:
> 
> * Rather than having the banner only appear on first visit of a project page or profile page (and disappear after navigating away and back to the page), I think it would be better to keep the banner present on subsequent visits to the page, and when a user clicks the "X" button, we then hide the banner for all subsequent visits
> * For  the "profile" banner, we should also hide the banner if a user has set the following profile features:
>   
>   * title & banner image & avatar & at least one external link (twitter, website, etc.)
> 
> @erikalogie I discussed this with @blushi earlier today, but would be good to hear your thoughts on this suggested change in functionality.

This sounds good to me @clevinson @blushi!

**Comment by blushi** [comment]
@erikalogie @clevinson this is ready for testing again with modified behavior

**Comment by erikalogie** [comment]
LGTM!

File: web-marketplace/src/lib/auth/auth.tsx [hunk 1]
```diff
@@ -56,6 +59,35 @@ export const AuthProvider: React.FC<React.PropsWithChildren<unknown>> = ({
...
+  }, [
+    activeAccount?.bgImage,
+    activeAccount?.id,
+    activeAccount?.image,
+    activeAccount?.name,
+    activeAccount?.twitterLink,
+    activeAccount?.websiteLink,
+    profileBannerCard,
+    setProfileBannerCard,
+  ]);
```
Comment (aaronc) [inline web-marketplace/src/lib/auth/auth.tsx:89]: Seems like a long dependency array. Not a blocker, but any way to simplify this?

**Review by aaronc — APPROVED** [review]
utACK

File: web-marketplace/src/lib/auth/auth.tsx (hunk 1 above)
Comment (blushi) [reply web-marketplace/src/lib/auth/auth.tsx:89]: I guess I could just extract the condition value in the `if` to some variable and have the useEffect depend on this variable to make it more readable

**Pull request merged**